### ISSUE TITLE
Can JavaScript interact with ::backdrop ?

### ISSUE THREAD

**Issue by DanKaplanSES** [mdn/content#27915]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/::backdrop

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/CSS/::backdrop#examples

### What information was incorrect, unhelpful, or incomplete?

I'm experimenting with modal \<dialog> boxes and I'm wondering how I can interact with `::backdrop` with JavaScript. Is it even possible? `document.querySelector("::backdrop")` always returns `null`, but it doesn't error like `document.querySelector(":foo")` does. I may be overthinking it, but I'm inferring that to mean there is some way to interact with it?

### What did you expect to see?

I'm used to be able to select pseudo elements with `querySelector` but I can't figure out how to do it with this one. I'm guessing that's because it can't be done, but I'd like to be able to confirm that with information on this page. If only CSS can interact with `::backdrop`, it would help if that was explicitly stated somewhere on the page. 

### Do you have any supporting links, references, or citations?

https://drafts.csswg.org/css-position-4/#backdrop

### Do you have anything more you want to share?

For additional context, I'm trying to figure out if there's a way to select the topmost modal `<dialog>` element without a knowing its `id` attribute or a unique `class` attribute. 

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/_doublecolon_backdrop`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/::backdrop
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/_doublecolon_backdrop/index.md
* Last commit: https://github.com/mdn/content/commit/559c464755abf4ca999a21ce7f3cb1027143f68f
* Document last modified: 2023-05-03T14:59:02.000Z

</details>

**Comment by OnkarRuikar** [comment]
We can not get handle to the pseudo elements, e.g. `::before`, using the Selectors API.
MDN
> CSS pseudo-elements will never return any elements, as specified in the Selectors API. [[ref]](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#usage_notes)

W3C
> NOTE: Authors are advised that while the use of pseudo-elements in selectors is permitted, they will not match any elements in the document, and thus would not result in any elements being returned.  [[ref]](https://www.w3.org/TR/selectors-api/#grammar)

But you can get references to pseudo classes, e.g. `:active`. Because the element is present in the markup.

As this is related to the Selectors API and has already been mentioned in https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#usage_notes section, there is no point in mentioning this DOM API fact on every CSS pseudo-element page.

***

> For additional context, I'm trying to figure out if there's a way to select the topmost modal `<dialog>` element without a knowing its id attribute or a unique class attribute.

You could try something like this:
```css
dialog:is(:modal, :focus, :fullscreen) {
  background: linear-gradient(lime, purple) !important;
}
````
Or
```css
dialog:has(#a-known-element) {
 background: red;
}
```

**Comment by Josh-Cena** [comment]
That's right. Pseudoelements aren't elements and what would the return type be should they be selectable anyway? They don't correspond to any interfaces. Plus there's no downside to adding an ID/class to your dialog and keeping track of which one is shown.

**Comment by DanKaplanSES** [comment]
Ah! So sorry. I confused pseudo-elements for pseudo-classes.

> there is no point in mentioning this DOM API fact on every CSS pseudo-element page

Completely agree. 

> You could try something like this:
> 
> ```css
> dialog:is(:modal, :focus, :fullscreen) {
>   background: linear-gradient(lime, purple) !important;
> }
> ```

This appears to select for all open modals instead of the topmost. I created in interactive example: https://jsfiddle.net/wk4gfznt/2/

**Comment by Josh-Cena** [comment]
I'm curious: under what circumstance would you have a bunch of modals open, but the JavaScript code is unaware of their order of opening? If you are using `show`/`showModal` to toggle them, then each time you can also record the dialog that currently emerges on top, and then listen for the `close` event to remove them from the stack.

**Comment by DanKaplanSES** [comment]
@Josh-Cena Oh I never _needed_ a selector to do this--in fact, I already keep track of the \<dialog> elements in an array like you suggested. I was just curious if a selector could do this.

**Comment by OnkarRuikar** [comment]
> I was just curious if a selector could do this.

None of [the pseudo-classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes#alphabetical_index) has such capability.

However, if you set `autofocus` on dialog elements then selector `dialog:has(:focus)` does what you want:
```html
<style>
  dialog:has(:focus) {
    background: linear-gradient(lime, aqua);
  }
</style>

<div>
  <dialog autofocus>
     a
  </dialog>
  <dialog autofocus>
     b
  </dialog>
</div>
```

Demo: https://jsfiddle.net/gzdc5sj9/1/
